lib/resources/etc_hosts.rb

```diff
@@ -0,0 +1,82 @@
+# encoding: utf-8
+# author: Matthew Dromazos
+
+require 'utils/parser'
+require 'pry'
+class EtcHosts < Inspec.resource(1)
+  name 'etc_hosts'
+  desc 'Use the etc_hosts InSpec audit resource to find an
+    ip_address and its associated hosts'
+  example "
+  describe etc_hosts.where { ip_address == '127.0.0.1' } do
+    its ( 'ip_address' ) { should eq ['127.0.0.1'] }
+    its ( 'canonical_hostname' ) { should eq ['localhost'] }
+    its ( 'all_host_names' ) { should eq [['localhost', 'localhost.localdomain', 'localhost4', 'localhost4.localdomain4']] }
+  end
+  "
+
+  attr_reader :params
+
+  include CommentParser
+
+  def initialize(hosts_path = nil)
+    return skip_resource 'The `etc_hosts` resource is not supported on your OS.' unless inspec.os.linux? || inspec.os.windows?
+    @conf_path      = get_hosts_path_by_os(hosts_path)
```

Comment:
It's more common to handle the override here, rather than in the `get_hosts_path_by_os` method...

```ruby
@conf_path = hosts_path || get_hosts_path_by_os
```

I'd also recommend naming the method `default_hosts_file_path` so it's clear why the method exists and its purpose.